text/0000-project-descriptor.md

```diff
@@ -0,0 +1,214 @@
+# Meta
+[meta]: #meta
+- Name: Minimal Project Descriptor
+- Start Date: 2019-06-11
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: https://github.com/buildpack/rfcs/pull/25
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a new file, `project.toml`, containing configuration for apps, services, functions, and buildpacks.
+
+# Motivation
+[motivation]: #motivation
+
+This proposal is meant to address all of the following issues:
+
+* Build metadata (i.e. information about the code being built)
+* [Ignoring files](https://github.com/buildpack/pack/issues/210)
+
+It also provides a foundation for solving the problems defined in the [Application descriptor issue](https://github.com/buildpack/spec/issues/44).
+
+# What it is
+[what-it-is]: #what-it-is
+
+Terminology:
+
+* **project**: a repository containing source code for an app, service, function, buildpack or a monorepo containing any combination of those.
+* **image**: the output of a running a buildpack(s) against a project
+
+The target personas for this proposal is buildpack users who need to enrich or configure buildpack execution. The elements in  `project.toml` support many different use cases including these top level tables:
+
+- `[project]`: (optional) defines configuration for a project
+- `[metadata]`: (optional) metadata about the repository
+
+Here is an overview of the complete schema:
+
+```toml
+[project]
+id = "<string>" # machine readble
+name = "<string>" # human readable
+version = "<string>"
+authors = ["<string>"]
+documentation-url = "<url>"
+source-url = "<url>"
+
+[[project.licenses]]
+type = "<string>"
+uri = "<uri>"
+
+[build]
+include = ["<string>"]
+exclude = ["<string>"]
+
+[[build.buildpacks]]
+id = "<string>"
+version = "<string>"
+uri = "<string>"
+
+[[build.env]]
+name = "<string>"
+value = "<string>"
+
+[metadata]
+# additional arbitrary keys allowed
+```
+
+The following sections describe each part of the schema in detail.
+
+## `[project]`
+
+The top-level `[project]` table may contain configuration about the repository, including `id` and `version`, but also metadata about how it is authored, documented, and version controlled.
+
+The `project.id`
+
+```toml
+[project]
+id = "<string>"
+name = "<string>"
+version = "<string>"
+```
+
+* `id` - the machine readable identifier of the project (ex. "com.example.myservice")
+* `name` - the human readable name of the project (ex. "My Example Service")
+* `version` - and arbitrary string representing the version of the project
+
+## `[project.licenses]`
+
+* `type` - This uses the [SPDX 2.1 license expression](https://spdx.org/spdx-specification-21-web-version). This matches the identifier and must be included in the [SPDX Lincenes List](https://spdx.org/licenses/). As of this writing, it's on version 3.7.
```

Comment:
```suggestion
* `type` - This uses the [SPDX 2.1 license expression](https://spdx.org/spdx-specification-21-web-version). This matches the identifier and must be included in the [SPDX Licenses List](https://spdx.org/licenses/). As of this writing, it's on version 3.7.
```

Reply:
I don't think that there should be a hard requirement that **only** contain SPDX values.  It's a good suggestion, but I think it's reasonable to use a value that's broadly applicable but not open source (e.g. Oracle BCL, Pivotal Proprietary).  This is something that you can map back to an automated value, but doesn't have an SPDX id.